shell.nix

```diff
@@ -0,0 +1,18 @@
+let _pkgs = import <nixpkgs> { };
```

Comment:
let's move this back into the `binary/` directory, update `IPXE_NIX_SHELL:=shell.nix` -> `IPXE_NIX_SHELL:=binary/shell.nix` in the `Makefile` and get this PR merged.

Reply:
I've put the shell.nix file back, still don't like it though.